King-Benx/android-level-up
# ft #156187450: Network Utility Class

#### What does this PR do?
This PR implements a network utility class to ensure network connectivity while using the application

#### Description of Task to be completed?
Initially, the application was operating on assumption that there is always an internet connection which was not realistic.
In order to improve the user experience and avoid the user from seeing a blank screen, this PR delivers a custom class called NetworkConnectivity that helps solve this issue, a SnackBar is also availed in order to help the user know what the problem is and help invoke a network connection if necessary.

#### WALKTHROUGH
![walkthrough_internet_connection](https://user-images.githubusercontent.com/17830204/44542456-8443fc80-a715-11e8-8e78-ab914cf965f6.gif)

#### How should this be manually tested?
1. Clone the repository onto your machine and build the app. 
2. Run the emulator.
3. Switch to airplane mode to ensure that there is no internet connection to the application.
4. Tap the application icon.
5. On the screen, a snackbar will present itself with an option to select a network connection.
6. Swipe down once you have a network connection in order to refresh the screen.

#### Background context
NONE

#### What are the relevant pivotal tracker stories?
[Delivers #157354404]
